.github/workflows/external-component-bot.yml

```
@@ -0,0 +1,154 @@
+name: Add External Component Comment
+
+on:
+  pull_request:
```

Without this, the token will not get permissions to write
```suggestion
  pull_request_target:
```

---

It seems to work as is

---

So `pull_request_target` implicitly grants fairly broad permissions. Using `pull_request` with explicit permissions is safer.

---

I think it is working for you as you have write permissions to this repo so it considers your PRs safe to be granted those higher permissions. If a contributor without write access makes a PR, it wont work. 

Think of it this way, if some Joe Public makes a PR and adds or modifies such a workflow, they can alter the permissions block and give their modified workflow unlimited access to the GH api. This is why `pull_request_target` exists, to give higher permissions, and PRs will only run with the workflow file that exists in the repos default branch and modifications are ignored in PRs